bolt-client/src/main.rs

```diff
@@ -0,0 +1,121 @@
+use alloy::{
+    eips::eip2718::Encodable2718,
+    hex,
+    network::{EthereumWallet, TransactionBuilder},
+    primitives::Address,
+    providers::{Provider, ProviderBuilder},
+    signers::local::PrivateKeySigner,
+};
+use beacon_api_client::mainnet::Client as BeaconApiClient;
+use clap::Parser;
+use eyre::{bail, Result};
+use tracing::info;
+use url::Url;
+
+mod registry;
+use registry::BoltRegistry;
+
+mod utils;
+use utils::*;
+
+#[derive(Parser)]
+struct Opts {
+    /// Bolt RPC URL to send requests to
+    #[clap(short = 'p', long, default_value = "http://135.181.191.125:8015/", env)]
+    rpc_url: Url,
+    /// Private key to sign transactions with
+    #[clap(short = 'k', long, env)]
+    private_key: String,
+    /// Optional nonce offset to use for the transaction
+    #[clap(short, long, default_value_t = 0, env)]
+    nonce_offset: u64,
+    /// Flag for generating a blob tx instead of a regular tx
+    #[clap(short, long, default_value_t = false)]
+    blob: bool,
+    /// Number of transactions to send in a sequence
+    #[clap(short, long, default_value_t = 1)]
+    count: u64,
+
+    /// Flag for using the registry to fetch the lookahead
+    #[clap(short, long, default_value_t = false)]
+    use_registry: bool,
```

Comment:
New `clap` trick of the day 😉 : if this is true, it requires both `registry_address` and `beacon_client_url`.
```suggestion
    /// Flag for using the registry to fetch the lookahead
    #[clap(short, long, default_value_t = false, requires_ifs([("true", "registry_address"), ("true", "beacon_client_url")]))]
)]
    use_registry: bool,
```

Reply:
I was missing "`clap` trick of the day" from my life. Thanks!